src/crypto/ShortHash.cpp

```diff
@@ -24,6 +24,16 @@ initialize()
     std::lock_guard<std::mutex> guard(gKeyMutex);
     crypto_shorthash_keygen(gKey);
 }
+
+std::array<unsigned char, crypto_shorthash_KEYBYTES>
```

Comment:
no need to make a copy here, just return a const ref

Reply:
We actually do need to copy here. `gKey` must be accessed only when `gKeyMutex` is locked, so we acquire the lock, make a copy, and return the copy. If we just return a direct reference to `gKey`, then we get race conditions and blow up. We also can't return a reference because we copy into a stack array and would return a stack reference. If your try to return a `const&` and return `std::move(arr)`, it works in `gcc` but not in `clang`.